taker-electron/src/main/endpoint.ts

```diff
@@ -0,0 +1,69 @@
+import { net } from "electron";
+import log from "electron-log";
+import nodenet from "node:net";
+
+const DEFAULT_PORT = 7113;
+
+export class Endpoint {
+    public url: string;
+
+    constructor(public port: number) {
+        this.url = `http://127.0.0.1:${port}`;
+    }
+
+    wait(onReady: (endpoint: string) => void) {
+        const alive = (timeout: number) => {
+            log.info(`Probing if ItchySats is alive at ${this.url}`);
+            const request = net.request(this.url);
+            request.on("response", () => onReady(this.url));
+            request.on("error", (error) => {
+                log.warn("Could not connect to ItchySats.");
+                log.error(error);
+                setTimeout(alive, timeout, timeout + timeout);
+            });
+            request.end();
+        };
+
+        setTimeout(alive, 500, 500, 500);
+    }
+
+    static create(): Promise<Endpoint> {
+        // retry checking random port by the given amount of max retries.
+        const retry = (
+            maxRetries: number,
+            fn: (port: number) => Promise<Endpoint>,
+            port: number,
+        ): Promise<Endpoint> => {
+            const minPort = 10_000;
+            const maxPort = 65_535;
+
+            return fn(port).catch(function(err) {
+                log.info(`Port: ${port} is not available retrying another random port. Retries: ${maxRetries}`);
+                if (maxRetries <= 0) {
+                    log.error(`Reached max amount of retries. quitting.`);
+                    throw err;
+                }
+                return retry(maxRetries - 1, fn, Math.floor(Math.random() * (maxPort - minPort + 1) + minPort));
```

Comment:
❓ Here we pass the function to be called and the value to be used in. Why this level of complexity? Are we planning to make the function that checks the available port to be configurable? 
A simple function call to `checkAvailablePort` should be good enough here or am I missing something?